contrib/scalar/scalar.c

```diff
@@ -1349,26 +1349,48 @@ static int cmd_reconfigure(int argc, const char **argv)
 	git_config(get_scalar_repos, &scalar_repos);
 
 	for (i = 0; i < scalar_repos.nr; i++) {
+		int failed = 0;
+		enum discovery_result discover_result;
 		const char *dir = scalar_repos.items[i].string;
 
 		strbuf_reset(&commondir);
 		strbuf_reset(&gitdir);
 
 		if (chdir(dir) < 0) {
 			warning_errno(_("could not switch to '%s'"), dir);
-			res = -1;
-		} else if (discover_git_directory(&commondir, &gitdir) < 0) {
-			warning_errno(_("git repository gone in '%s'"), dir);
-			res = -1;
-		} else {
-			git_config_clear();
+			failed = -1;
+			goto loop_end;
+		}
+
+		discover_result = discover_git_directory_reason(&commondir, &gitdir);
+
+		switch (discover_result) {
```

Comment:
Now that I look at it here in the diff, I could just put the method call in the switch.